feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/balance/UnbondingsAdapter.kt

```diff
@@ -30,13 +33,33 @@ class UnbondingsAdapter : ListAdapter<UnbondingModel, UnbondingsHolder>(Unbondin
 }
 
 class UnbondingsHolder(override val containerView: View) : RecyclerView.ViewHolder(containerView), LayoutContainer {
+    var timer: CountDownTimer? = null
 
     fun bind(unbonding: UnbondingModel) = with(containerView) {
         with(unbonding) {
+            if (timer != null) timer?.cancel()
+
+            timer = object : CountDownTimer(timeLeft, 1000) {
+                override fun onTick(millisUntilFinished: Long) {
+                    val days = millisUntilFinished.getDays()
+
+                    itemListElementDescriptionLeft.text = if (days > 0)
+                        resources.getQuantityString(R.plurals.staking_payouts_days_left, days, days)
+                    else
+                        millisUntilFinished.formatTime()
+                }
+
+                override fun onFinish() {
+                    itemListElementDescriptionLeft.text = 0L.formatTime()
+
+                    cancel()
+                }
+            }
+
+            timer?.start()
```

Comment:
Duplicated code with above. Suggest to move it to extension function like "textView.startTimer(time: Long)". To cache timer `view.setTag(key, value)` can be used

Reply:
Done